### PR TITLE
UCP/API: Added name to worker parameters.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Added worker name to UCP worker parameters to identify the worker.
Added worker name to UCP worker attributes to get the actual worker name.

## Why ?
The parameter is required for more convenient run time analysis of UCX-based application. E.g. ucx_vfs tool.

### CODE REVIEW

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Comment by yosefe** [comment]
@shamisp @tonycurtis can you pls take a look?

**Review by tonycurtis — APPROVED** [review]

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -1188,6 +1196,18 @@ typedef struct ucp_worker_params {
      */
     uint64_t                flags;
 
+    /**
+     * Worker name is intended for identification of the worker during tracing
```
Comment (shamisp) [inline src/ucp/api/ucp.h:1200]: IMHO the first sentence has to be rewritten. @tonycurtis  I don't like the passive form. `The worker name is an id of the worker object for the tracing tool`?

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -1105,6 +1107,12 @@ typedef struct ucp_worker_attr {
      * Maximal allowed header size for @ref ucp_am_send_nbx routine
      */
     size_t                max_am_header;
+
+    /**
+     * The name is intended for identification of the worker during tracing and
```
Comment (shamisp) [inline src/ucp/api/ucp.h:1112]: No need  for passive form here. The name is the ID of the worker. The tracing tool uses the ID to identity an worker object.

File: src/ucp/api/ucp.h [hunk 3]
```diff
@@ -1188,6 +1196,18 @@ typedef struct ucp_worker_params {
      */
     uint64_t                flags;
 
+    /**
+     * Worker name is intended for identification of the worker during tracing
+     * and analysis of UCX-based applications.
+     * This value is optional.
+     * The actual name of the worker can be obtained by @ref ucp_worker_query
```
Comment (shamisp) [inline src/ucp/api/ucp.h:1203]: The library can change the name provided by user in order to avoid potential name conflicts. It is advise to query the name using @ref ucp_worker_query

**Review by shamisp — CHANGES_REQUESTED** [review]
@tonycurtis can you please help improve documentation for this one. thanks

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (rakhmets) [reply src/ucp/api/ucp.h:1200]: Tracing and analysis tools can identify the worker using the name.

@shamisp @tonycurtis 
How about that? I tried to make it a little more concise.

File: src/ucp/api/ucp.h (hunk 2 above)
Comment (rakhmets) [reply src/ucp/api/ucp.h:1112]: Tracing and analysis tools can identify the worker using the name.

The same as above.

File: src/ucp/api/ucp.h (hunk 3 above)
Comment (rakhmets) [reply src/ucp/api/ucp.h:1203]: The library can change the name provided by a user to avoid potential name conflicts. Please use `@ref  ucp_worker_query` to get the actual worker name.

@shamisp @tonycurtis is this one ok?

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (tonycurtis) [reply src/ucp/api/ucp.h:1200]: Maybe "this name", but otherwise, yes.

File: src/ucp/api/ucp.h (hunk 2 above)
Comment (shamisp) [reply src/ucp/api/ucp.h:1112]: @rakhmets my main concern with the original documentation is `The name is intended ` which is passive form and hard to parse.

**Comment by shamisp** [comment]
@tonycurtis - can you please help us make it more readable. Please see my above comments.

File: src/ucp/api/ucp.h [hunk 4]
```diff
@@ -1188,6 +1195,17 @@ typedef struct ucp_worker_params {
...
     uint64_t                flags;
 
+    /**
+     * Tracing and analysis tools can identify the worker using this name.
+     * This value is optional.
+     * The worker name may differ from the name specified in the parameters to
+     * avoid potential name conflicts. Please use @ref ucp_worker_query to get
+     * the actual name of the worker.
+     * If @ref UCP_WORKER_PARAM_FIELD_NAME is not set in the field_mask, the
+     * worker will have a default name.
```
[the remark this answers is not in the document]
Comment (tonycurtis) [reply src/ucp/api/ucp.h:1205]: rework of lines 1198-1205:

Tracing and analysis tools can identify the worker using this name.  To retrieve the worker's name, use @ref ucp_worker_query, as the name you supply may be changed by UCX under some circumstances, e.g. a name conflict.  This field is only assigned if you set @ref UCP_WORKER_PARAM_FIELD_NAME in the field mask.  If not, then a default unique name will be created for you.

File: src/ucp/core/ucp_context.h [hunk 5]
```diff
@@ -260,7 +260,7 @@ typedef struct ucp_context {
     /* All configurations about multithreading support */
```
Comment (tonycurtis) [inline src/ucp/core/ucp_context.h:260]: I don't understand the comment.  Is something like

/* Configuration of multi-threadiing support */

meant?

File: src/ucp/api/ucp.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (tonycurtis) [reply src/ucp/api/ucp.h:1205]: The comment in ucp_params_t for the name field needs to be revisited too

**Review by tonycurtis — CHANGES_REQUESTED** [review]

File: src/ucp/core/ucp_context.h (hunk 5 above)
Comment (rakhmets) [reply src/ucp/core/ucp_context.h]: Done.

File: src/ucp/api/ucp.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (rakhmets) [reply src/ucp/api/ucp.h]: Updated description here and in ucp_params_t.

File: src/ucp/api/ucp.h [hunk 6]
```diff
@@ -1054,7 +1056,7 @@ typedef struct ucp_context_attr {
      * The name is intended for identification of context during tracing and
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply src/ucp/api/ucp.h]: Done.

File: src/ucp/api/ucp_def.h [hunk 7]
```diff
@@ -749,10 +749,10 @@ typedef struct ucp_ep_params {
 
 /**
  * @ingroup UCP_CONTEXT
- * @brief Maximum size of the context name in @ref ucp_context_attr_t provided
- * by @ref ucp_context_query.
+ * @brief Maximum size of the UCP entity name in structure of entity attributes
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply src/ucp/api/ucp_def.h:752]: Ok, maximum is used here. I will use this in the future

File: src/ucp/api/ucp.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (rakhmets) [reply src/ucp/api/ucp.h]: We agreed to not describe the possible name changes in different cases in this discussion https://github.com/openucx/ucx/pull/6368#discussion_r582145059
... to avoid creating dependency on implementation. User can get the actual name using query method which is mentioned in the description.

**Comment by rakhmets** [comment]
@shamisp @tonycurtis
could you please to approve the changes if everything is ok

**Review by tonycurtis — APPROVED** [review]
Let's address the use of maximal/maximum in a separate issue

**Review by yosefe — APPROVED** [review]

**Comment by brminich** [comment]
@shamisp, can you please review the latest changes?

**Review by shamisp — APPROVED** [review]
